### ISSUE TITLE
CLI doesn't include styles or transformations from `<style>` tag in HTML files

### ISSUE THREAD

**Issue by hrishikesh-k** [unocss/unocss#2445]
### UnoCSS version

0.50.6

### Describe the bug

Discussed here: https://github.com/unocss/unocss/discussions/2380

TL;DR: When using the CLI, along with `transformDirectives`, the styles added to `<style>` tag in HTML are not transformed. The same CSS from the `<style>` tag, when put in a separate CSS file, works perfectly fine.

WindiCSS had a CLI flag `--style` that would include styles from `<style lang="windi">` blocks. UnoCSS doesn't seem to have a similar functionality (or it at least doesn't seem to be documented).

### Reproduction

[unocss-cli-test.zip](https://github.com/unocss/unocss/files/11241312/unocss-cli-test.zip)

### System Info

Node 18.1.5.0
Chrome 112

### Validations

- [X] Read the [Contributing Guidelines](https://github.com/unocss/unocss/blob/main/CONTRIBUTING.md).
- [X] Check that there isn't [already an issue](https://github.com/unocss/unocss/issues) that reports the same bug to avoid creating a duplicate.
- [X] Check that this is a concrete bug. For Q&A open a [GitHub Discussion](https://github.com/unocss/unocss/discussions) or join our [Discord Chat Server](https://chat.antfu.me).
- [X] The provided reproduction is a [minimal reproducible example](https://stackoverflow.com/help/minimal-reproducible-example) of the bug.

**Comment by Hrishikesh-K** [comment]
Sorry, pinging again because Stalebot, but the issue is still relevant

**Comment by Simon-He95** [comment]
<img width="1043" alt="image" src="https://github.com/unocss/unocss/assets/57086651/b0f5dd04-e598-430b-971c-3000aab94c59">
remove the `strict` option will work
